File: samples/VisualBasic/Northwind/CustomerCustomerDemo.vb
Imports System
Imports DbExtensions

Namespace Northwind

   <Table>
   Public Class CustomerCustomerDemo

      <Column(IsPrimaryKey:=True)>
      Property CustomerID As String

      <Column(IsPrimaryKey:=True)>
      Property CustomerTypeID As String

      <Association(ThisKey:=NameOf(CustomerID))>
      Property Customer As Customer

      <Association(ThisKey:=NameOf(CustomerTypeID))>
      Property CustomerDemographic As CustomerDemographic

   End Class

End Namespace

File: samples/CSharp/Northwind/CustomerCustomerDemo.cs
using System;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table]
   public class CustomerCustomerDemo {

      [Column(IsPrimaryKey = true)]
      public string CustomerID { get; set; }

      [Column(IsPrimaryKey = true)]
      public string CustomerTypeID { get; set; }

      [Association(ThisKey = nameof(CustomerTypeID))]
      public CustomerDemographic CustomerDemographic { get; set; }

      [Association(ThisKey = nameof(CustomerID))]
      public Customer Customer { get; set; }
   }
}
